src/java.security.jgss/share/classes/sun/security/krb5/internal/KDCRep.java

```diff
@@ -138,20 +138,24 @@ protected void init(DerValue encoding, int req_type)
                         " req type is " + req_type);
 
                 System.out.println(">>> KDCRep: Message in bytes is =>");
-                byte[] dataBytes = encoding.getDataBytes();
-                StringBuilder sb = new StringBuilder();
-                for (int i = 0; i < dataBytes.length; i++) {
-                    if ((i % 16) == 0) {
-                        sb.append(String.format("%06X", i));
+                byte[] dataBytes = encoding.getData().toByteArray();
+                if (dataBytes != null) {
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 0; i < dataBytes.length; i++) {
+                        if ((i % 16) == 0) {
+                            sb.append(String.format("%06X", i));
+                        }
+                        sb.append(String.format(" %02X", dataBytes[i] & 0xFF));
+                        if ((i % 16) == 15) {
+                            System.out.println(sb.toString());
+                            sb.setLength(0);
+                        }
                     }
-                    sb.append(String.format(" %02X", dataBytes[i] & 0xFF));
-                    if ((i % 16) == 15) {
-                        System.out.println(sb.toString());
-                        sb.setLength(0);
-                    }
-                }
-                if (sb.length() > 0) {
+                    if (sb.length() > 0) {
                     System.out.println(sb.toString());
+                    }
+                } else {
+                    System.out.println("\tReply message is empty.");
```

Comment:
I don't think this difference (as compared with newer versions) is necessary.

Reply:
Done.